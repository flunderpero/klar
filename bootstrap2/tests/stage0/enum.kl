enum Color:
    Red
    Green
    Blue
    HSL(hue i32, saturation i32, lightness i32)
end

fn print_color(c Color):
    if c == Color.Blue:
        print("Blue")
        return
    end
    if c == Color.Green:
        print("Green")
        return
    end
    if c == Color.Red:
        print("Red")
        return
    end
    print("HSL")
end

fn main():
    let color = Color.Blue
    if color == Color.Blue:
        print("Ok: Blue")
    else:
        print("Fail: Not blue")
    end
    let orange = Color.HSL(30, 100, 50)
    print(f"Orange: H{orange.hue}, S{orange.saturation}, L{orange.lightness}")
    print_color(Color.Blue)
    print_color(Color.Red)
    print_color(Color.Green)
    print_color(orange)
end
