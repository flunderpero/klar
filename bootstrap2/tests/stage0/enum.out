Ok: Blue
Orange: H30, S100, L50
Blue
Red
Green
HSL
